src/optimagic/deprecations.py

```diff
@@ -325,6 +325,15 @@ def throw_key_warning_in_derivatives():
     warnings.warn(msg, FutureWarning)
 
 
+def throw_dict_constraints_future_warning():
+    msg = (
+        "Specifying constraints as a dictionary is deprecated and will be removed in "
+        "optimagic version 0.6.0. Please use a constraint object imported from "
+        "optimagic.constraints instead."
```

Comment:
This should take constraints as argument and give a more constructive error message. Similar to what we do [here](https://github.com/OpenSourceEconomics/optimagic/blob/19029f53dfe4a599bb4f121eea3ed3fabf54f3a3/src/optimagic/deprecations.py#L169) but only warn, not replace.

Reply:
Then the control flow to check whether a warning is necessary can also go into the function.